sunpy/map/maputils.py

```diff
@@ -164,11 +166,73 @@ def solar_angular_radius(coordinates):
     return sun._angular_radius(coordinates.rsun, coordinates.observer.radius)
 
 
-def sample_at_coords(smap, coordinates):
+def get_neighboring_indices(indices, position, dimensions):
     """
-    Samples the data in a map at given series of coordinates.
-    Uses nearest-neighbor interpolation of coordinates in map, as
-    it effectively uses array indexing.
+    Get the neighboring indices relative to given indices.
+
+    Parameters
+    ----------
+    indices : 2-D ndarray of integers with shape (2, n)
+        Indices to find neighbors for.
+    position : 1-D ndarray of shape (2,) or 2-element list
+        x and y position of neighbor relative to given indices.
```

Comment:
This API seems like awkward and prone to mistakes.  Unless you envision asymmetric use cases, I think the API should instead by specifying the radius in pixels of the neighborhood to return.

Reply:
Yes, I agree. Specifying the radius is probably the better option. I'll change it

Reply:
Latest commit exposes the pixel radius.